test/services/active_challenges_cache.js

```diff
@@ -0,0 +1,213 @@
+      expect(activeChallengesCache.has('0xc0ffee2')).to.be.true;
+    });
+  });
+
+  describe('Decrease active count', () => {
+    beforeEach(() => {
+      activeChallengesCache.add(exampleChallenge);
+      activeChallengesCache.add(exampleChallenge2);
+    });
+
+    it('decreases active count for different challenges', () => {
+      activeChallengesCache.decreaseActiveCount({challengeId: '0xc0ffee'});
+      expect(activeChallengesCache.get('0xc0ffee')).to.deep.equal({
+        sheltererId: '0xbeef',
+        bundleId: '0xdeadbeef',
+        count: 11,
+        blockNumber: 1,
+        logIndex: 0
+      });
+      expect(activeChallengesCache.get('0xc0ffee2')).to.deep.equal({
+        sheltererId: '0xbeef2',
+        bundleId: '0xdeadbeef2',
+        count: 43,
+        blockNumber: 2,
+        logIndex: 1
+      });
+    });
+
+    it('decreases active count for same challenge many times', () => {
+      activeChallengesCache.decreaseActiveCount({challengeId: '0xc0ffee'});
+      activeChallengesCache.decreaseActiveCount({challengeId: '0xc0ffee'});
+      activeChallengesCache.decreaseActiveCount({challengeId: '0xc0ffee'});
+      expect(activeChallengesCache.get('0xc0ffee')).to.deep.equal({
+        sheltererId: '0xbeef',
+        bundleId: '0xdeadbeef',
+        count: 9,
+        blockNumber: 1,
+        logIndex: 0
+      });
+    });
+
+    it('does not throw if tried to decreases active count of not existing challenge', () => {
+      activeChallengesCache.decreaseActiveCount([{challengeId: '0xdeadface'}]);
+      expect(activeChallengesCache.has('0xc0ffee')).to.be.true;
+      expect(activeChallengesCache.has('0xc0ffee2')).to.be.true;
+    });
+
+    it('removes challenge when active count falls to 0', () => {
+      for (let ind = 0; ind < 12; ind++) {
+        activeChallengesCache.decreaseActiveCount({challengeId: '0xc0ffee'});
+      }
+      expect(activeChallengesCache.has('0xc0ffee')).to.be.false;
+    });
+  });
+
+  describe('Active challenges getter', () => {
+    beforeEach(() => {
+      activeChallengesCache.add(exampleChallenge);
+      activeChallengesCache.add(exampleChallenge2);
+      activeChallengesCache.add(exampleChallenge3);
+    });
+
+    it('returns all active challanges as an array', () => {
+      expect(activeChallengesCache.activeChallenges).to.deep.equal([exampleChallenge, exampleChallenge3, exampleChallenge2]);
+    });
+  });
+
+  describe('Sorts chronogically', () => {
+    const events = [
+      {blockNumber: 5, logIndex: 2},
+      {blockNumber: 9, logIndex: 4},
+      {blockNumber: 5, logIndex: 7}
+    ];
+
+    it('in increasing blockNumber and logIndex order', () => {
+      expect(activeChallengesCache.sortChronologically(events)).to.deep.equal([{blockNumber: 5, logIndex: 2}, {blockNumber: 5, logIndex: 7}, {blockNumber: 9, logIndex: 4}]);
+    });
+  });
+
+  describe('Applies multiple changes', () => {
+    it('Applies in correct order', () => {
```

Comment:
This test is very hard on the reader. You can't quickly figure out why the expected results are what they are. Consider splitting into a couple of smaller tests.

Reply:
To be done.